### PR TITLE
Chore/update to godot 4.2

### PR DESCRIPTION
**Pull request by piiertho**
This updates module to `4.2-stable`.  
When this is merged, we can update #534 and #498 PRs.

### CODE REVIEW

**Review by CedNaru — APPROVED** [review]

File: docs/src/doc/user-guide/api-differences.md [hunk 1]
```diff
@@ -97,6 +97,19 @@ In Kotlin, they are available inside the `GD` singleton. Don't forget that some
 
 For comfort, some Objects got some additional functions to enjoy some Kotlin syntax sugar. You can find them all [in this folder](https://github.com/utopia-rise/godot-kotlin-jvm/tree/master/kt/godot-library/src/main/kotlin/godot/extensions)
 
+## Notifications
```
Comment (chippmann) [inline docs/src/doc/user-guide/api-differences.md:100]: why is this added again? wasn't this already part of the other pr?

**Review by chippmann — APPROVED** [review]
